src/test/java/com/jakewharton/rxrelay2/ReplayRelayBoundedConcurrencyTest.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package com.jakewharton.rxrelay2;
+
+import io.reactivex.Scheduler;
+import io.reactivex.observers.DefaultObserver;
+import io.reactivex.schedulers.Schedulers;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ReplayRelayBoundedConcurrencyTest {
+
+
+    @Test
+    public void testReplaySubjectEmissionSubscriptionRace() throws Exception {
```

Comment:
Hm, loks like there is a leak somewhere here:

`Java HotSpot(TM) 64-Bit Server VM warning: INFO: os::commit_memory(0x00007fc588c2b000, 12288, 0) failed; error='Cannot allocate memory'`

Builds on both [JDK7](https://travis-ci.org/JakeWharton/RxRelay/jobs/178206614#L271) and [JDK8](https://travis-ci.org/JakeWharton/RxRelay/jobs/178206615#L271) failed for same reason.